src/v/config/property.h

```diff
@@ -408,6 +461,74 @@ inline binding<T> mock_binding(T&& value) {
     return binding<T>(std::forward<T>(value));
 }
 
+/**
+ * A conversion property binding contains the result of application of
+ * a conversion function to property value. The result is update in-place
+ * whenever the property is updated in the cluster configuration.
+ *
+ * This is useful when the actually useful value of the configuration property
+ * needs to be different from its representation in the cluster config.
+ *
+ * /param U The type of converted value, accessible from the binding
+ * /param T The type of the parent property
+ */
+template<typename U, typename T>
+class conversion_binding : public binding_base<T> {
+public:
+    using conversion_func = std::function<U(const T&)>;
+
+private:
+    U _value;
+    conversion_func _convert;
+    std::optional<std::function<void()>> _on_change;
```

Comment:
This `_on_change` declaration looks to be redundant (binding_base has _on_change)